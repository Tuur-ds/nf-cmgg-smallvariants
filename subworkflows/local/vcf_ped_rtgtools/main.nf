//
// ADD PED HEADER
//

include { RTGTOOLS_PEDFILTER    } from '../../../modules/nf-core/rtgtools/pedfilter/main'
include { BCFTOOLS_ANNOTATE     } from '../../../modules/nf-core/bcftools/annotate/main'

workflow VCF_PED_RTGTOOLS {
    take:
        ch_vcfs // channel: [mandatory] [ val(meta), path(vcf), path(tbi) ] => The post-processed VCFs
        ch_peds // channel: [mandatory] [ val(meta), path(peds) ] => The PED files retrieved from SOMALIER_RELATE

    main:

    def ch_versions = Channel.empty()

    //
    // Remove extra columns from the samples TSV and convert to a VCF header
    //

    RTGTOOLS_PEDFILTER(
        ch_peds
    )
    ch_versions = ch_versions.mix(RTGTOOLS_PEDFILTER.out.versions.first())

    //
    // Add the PED headers to the VCF using bcftools annotate --header-lines
    //

    def ch_annotate_input = ch_vcfs
        .join(RTGTOOLS_PEDFILTER.out.output, failOnDuplicate:true, failOnMismatch:true)
        .map { meta, vcf, _tbi, ped_vcf ->
            [ meta, vcf, [], [], [], ped_vcf ]
        }

    BCFTOOLS_ANNOTATE(
        ch_annotate_input
    )
    ch_versions = ch_versions.mix(BCFTOOLS_ANNOTATE.out.versions.first())

    emit:
    ped_vcfs = BCFTOOLS_ANNOTATE.out.vcf    // [ val(meta), path(vcf) ]
    versions = ch_versions                  // [ path(versions) ]
}
