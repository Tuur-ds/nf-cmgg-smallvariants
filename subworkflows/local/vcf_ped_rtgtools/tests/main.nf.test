nextflow_workflow {

    name "Test Workflow VCF_PED_RTGTOOLS"
    script "../main.nf"
    workflow "VCF_PED_RTGTOOLS"

    tag "subworkflows"
    tag "subworklows_local"
    tag "vcf_ped_rtgtools"

    test("vcf_ped_rtgtools - default") {

        when {
            params {
                add_ped = true
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"Ashkenazim", family:"Ashkenazim", family_samples:"NA24143,NA24149,NA24385", caller:"haplotypecaller"],
                    file(params.famvcf, checkIfExists:true),
                    file(params.famtbi, checkIfExists:true)
                ])
                input[1] = Channel.of([
                    [id:"Ashkenazim", family:"Ashkenazim", family_samples:"NA24143,NA24149,NA24385", caller:"haplotypecaller"],
                    file(params.ped, checkIfExists:true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.ped_vcfs.collect { [it[0], "variantsMD5:${path(it[1]).vcf.variantsMD5}"] }
                ).match() }
            )
        }

    }

}
