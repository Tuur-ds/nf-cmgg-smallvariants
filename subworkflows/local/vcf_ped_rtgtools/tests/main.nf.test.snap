{
    "vcf_ped_rtgtools - default": {
        "content": [
            [
                [
                    {
                        "id": "Ashkenazim",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143,NA24149,NA24385",
                        "caller": "haplotypecaller"
                    },
                    "variantsMD5:4367d1c96e701a1cc1a96b615381b278"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-14T15:54:56.709943272"
    }
}